test/libsolidity/syntaxTests/imports/shadowing_builtins_with_alias.sol

```diff
@@ -0,0 +1,7 @@
+==== Source: B.sol ====
+contract C {}
+==== Source: b ====
+pragma solidity >=0.0;
+import {C as msg} from "B.sol";
+// ----
+// Warning: (B.sol:0-13): This declaration shadows a builtin symbol.
```

Comment:
The source locations here are in fact a bit off - I created #7273 to track that. I confirmed that it's not related to this PR, but also happens exactly like this with two "proper" source files.